demos/guide/quic-hq-interop-server.c

```diff
@@ -375,10 +446,31 @@ static void process_new_stream(SSL *stream)
     size_t bytes_written = 0;
     size_t offset = 0;
     int rc;
+    int ret;
+    size_t total_read = 0;
 
     memset(buf, 0, BUF_SIZE);
-    if (SSL_read_ex(stream, buf, sizeof(buf) - 1, &nread) <= 0)
-        return;
+    for (;;) {
+        nread = 0;
+        ret = SSL_read_ex(stream, &buf[total_read],
+                          sizeof(buf) - total_read - 1, &nread);
```

Comment:
Can this be just `sizeof(buf) - total_read`?

Reply:
I suppose not since we need to leave space for the NUL terminator

Reply:
When/where do we write the NUL terminator? I assume it would be included in the call to SSL_read

Reply:
3 lines up, when we memset the entire buffer to 0.

Reply:
> When/where do we write the NUL terminator? I assume it would be included in the call to SSL_read

SSL_read reads binary data. It does not include a NUL terminator unless the peer sent it with a NUL terminator.